### ISSUE TITLE
[C6] Bare `except Exception:` swallows KeyboardInterrupt and MemoryError

### ISSUE THREAD

**Issue by imaddde867** [imaddde867/IPKE#40]
## Problem

Three files use `except Exception:` instead of specific exception types. While all instances carry `# noqa: BLE001` (showing awareness), `Exception` still catches `KeyboardInterrupt`, `SystemExit`, and `MemoryError` on Python < 3.8 semantics, and more importantly hides unexpected failures:

| File | Lines | Context |
|---|---|---|
| `src/processors/chunkers/breakpoint.py` | 37, 45 | Device resolution (CUDA/MPS availability) |
| `src/ai/knowledge_engine.py` | 342, 351 | GPU backend detection |
| `src/evaluation/metrics.py` | 73, 82 | spaCy model loading / fallback |

The device-resolution excepts in `breakpoint.py` and `knowledge_engine.py` silently fall back to `cpu` even on `MemoryError` — a GPU OOM during torch import would be masked.

## Fix

Replace with specific exception types appropriate to the context:

- Device resolution (torch import): `except (ImportError, RuntimeError)`
- spaCy loading: `except (OSError, ValueError)`

Log the exception message at `WARNING` level rather than silently passing.

Example:
```python
try:
    import torch
    if torch.cuda.is_available():
        return "cuda"
except (ImportError, RuntimeError) as exc:
    LOGGER.warning("CUDA check failed, falling back to CPU: %s", exc)
```

## Files

- `src/processors/chunkers/breakpoint.py:37`, `45`
- `src/ai/knowledge_engine.py:342`, `351`
- `src/evaluation/metrics.py:73`, `82`
